marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/saveSavedQuery.sjs

```diff
@@ -43,7 +43,15 @@ if (queryDocument.savedQuery.propertiesToDisplay == null || queryDocument.savedQ
     ds.throwBadRequest("Entity type properties to be displayed cannot be empty");
 }
 
-let id = queryDocument.savedQuery.id;
+const id = queryDocument.savedQuery.id;
+const positiveQuery = cts.andQuery([cts.collectionQuery("http://marklogic.com/data-hub/saved-query"), cts.jsonPropertyValueQuery("name", queryDocument.savedQuery.name),
+    cts.jsonPropertyValueQuery("owner", xdmp.getCurrentUser())]);
+const negativeQuery = cts.documentQuery("/saved-queries/" + id + ".json");
+const queryNameExists = cts.exists(cts.andNotQuery(positiveQuery, negativeQuery));
+if(queryNameExists) {
+    ds.throwBadRequest(`A query already exists with a name of ${queryDocument.savedQuery.name}`);
```

Comment:
I think this message needs to capture the fact that the current user already has the query; otherwise, the user could interpret this message as implying that no two users can use the same named queries. 

I recommend: "You already have a saved query with a name of ${the name}".